rotkehlchen/exchanges/kucoin.py

```diff
@@ -0,0 +1,775 @@
+                )
+                self.msg_aggregator.add_error(
+                    'Failed to deserialize a kucoin account. Ignoring it.',
+                )
+                continue
+            except (UnknownAsset, UnsupportedAsset) as e:
+                asset_tag = 'unknown' if isinstance(e, UnknownAsset) else 'unsupported'
+                self.msg_aggregator.add_warning(
+                    f'Found {asset_tag} kucoin asset {e.asset_name} while deserializing '
+                    f'an account. Ignoring it.',
+                )
+                continue
+            try:
+                usd_price = Inquirer().find_usd_price(asset=asset)
+            except RemoteError:
+                self.msg_aggregator.add_error(
+                    f'Failed to deserialize a kucoin account balance after failing to '
+                    f'request the USD price of {asset.identifier}. Ignoring it.',
+                )
+                continue
+
+            assets_balance[asset] += Balance(
+                amount=amount,
+                usd_value=amount * usd_price,
+            )
+
+        return dict(assets_balance)
+
+    def _check_for_system_clock_not_synced_error(
+            self,
+            error_code: Optional[int] = None,
+    ) -> None:
+        if error_code == API_SYSTEM_CLOCK_NOT_SYNCED_ERROR_CODE:
+            raise SystemClockNotSyncedError(
+                current_time=str(datetime.now()),
+                remote_server=f'{self.name}',
+            )
+
+    @staticmethod
+    def _deserialize_asset_movement(
+            raw_result: Dict[str, Any],
+            category_case: Literal['deposit', 'withdrawal'],
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> Tuple[Optional[AssetMovement], Optional[SkipReason]]:
+        """Process an asset movement result and deserialize it
+
+        May raise DeserializationError
+        """
+        if category_case == 'deposit':
+            category = AssetMovementCategory.DEPOSIT
+        elif category_case == 'withdrawal':
+            category = AssetMovementCategory.WITHDRAWAL
+        else:
+            raise AttributeError(f'Unexpected category case: {category_case}')
+
+        try:
+            timestamp = deserialize_timestamp(int(raw_result['createdAt'] / 1000))
+            if timestamp > end_ts:
+                return None, SkipReason.ABOVE_TIMESTAMP_RANGE
+            if timestamp < start_ts:
+                return None, SkipReason.BELOW_TIMESTAMP_RANGE
+
+            is_inner = raw_result['isInner']
+            if is_inner is True:
+                return None, SkipReason.INNER_MOVEMENT
+
+            address = raw_result['address']
+            transaction_id = raw_result['walletTxId']
+            amount = deserialize_asset_amount(raw_result['amount'])
+            fee = deserialize_fee(raw_result['fee'])
+            fee_currency_symbol = raw_result['currency']
+            # NB: id only exists for withdrawals
```

Comment:
Deposits do not have ID using this endpoint.
Besides it is not clear for me if `walletTxId` contains transaction hashes of blockchains or something internal from KuCoin. I've asked them, no answer yet.

Reply:
According to the docs and other python clients, the field will always be there and won't be None. They have not answered me yet (after 2 days) about its meaning though.

Reply:
I think here it probably makes sense to be more defensive. go a raw_result('id', '') since anyway the `link` is mostly informative in our trades and if it's missing it does not matter to us or to the user.